spec-main/api-context-bridge-spec.ts

```diff
@@ -60,1165 +60,2149 @@ describe('contextBridge', () => {
+                      }
+                    };
+                  }
+                });
+              });
+              const result = await callWithBindings(async (root: any) => {
+                return root.thing.foo.bar;
+              });
+              expect(result).to.equal('hi there');
+            });
+
+            it('should work with setters', async () => {
+              await makeBindingWindow(() => {
+                let a: any = null;
+                contextBridge.internalContextBridge!.overrideGlobalValueWithDynamicPropsFromIsolatedWorld(['thing'], {
+                  get foo () {
+                    return a;
+                  },
+                  set foo (arg: any) {
+                    a = arg + 1;
+                  }
+                });
+              });
+              const result = await callWithBindings(async (root: any) => {
+                root.thing.foo = 123;
+                return root.thing.foo;
+              });
+              expect(result).to.equal(124);
+            });
+
+            it('should work with nested getter / setter combos', async () => {
+              await makeBindingWindow(() => {
+                let a: any = null;
+                contextBridge.internalContextBridge!.overrideGlobalValueWithDynamicPropsFromIsolatedWorld(['thing'], {
+                  get thingy () {
+                    return {
+                      get foo () {
+                        return a;
+                      },
+                      set foo (arg: any) {
+                        a = arg + 1;
+                      }
+                    };
+                  }
+                });
+              });
+              const result = await callWithBindings(async (root: any) => {
+                root.thing.thingy.foo = 123;
+                return root.thing.thingy.foo;
+              });
+              expect(result).to.equal(124);
+            });
+
+            it('should work with deep properties', async () => {
+              await makeBindingWindow(() => {
+                contextBridge.internalContextBridge!.overrideGlobalValueWithDynamicPropsFromIsolatedWorld(['thing'], {
+                  a: () => ({
+                    get foo () {
+                      return 'still here';
+                    }
+                  })
+                });
+              });
+              const result = await callWithBindings(async (root: any) => {
+                return root.thing.a().foo;
+              });
+              expect(result).to.equal('still here');
+            });
+          });
         });
-        expect(result).to.equal('i-resolved');
       });
+    };
 
-      it('should proxy promises and reject with the correct value', async () => {
-        await makeBindingWindow(() => {
-          contextBridge.exposeInMainWorld('example', Promise.reject(new Error('i-rejected')));
+    generateTests(true);
+    generateTests(false);
+  });
+
+  describe('exposeInIsolatedWorld', () => {
```

Comment:
Given the similarity in implementation, I'd be in favor of only adding a simple test for `exposeInIsolatedWorld` rather than duplicating all of those for `exposeInMainWorld`.

This would cut down on testing time and redundancy.